Better error message for golang non-concrete types

Currently I get an error like:

panic: type value main.Foo must be a concrete type

It would be helpful to tell me the name of the field in Foo that's causing the problem.

Imported from Jira [BEAM-5199](https://issues.apache.org/jira/browse/BEAM-5199). Original Jira may contain additional context.
Reported by: ribrdb.

--- a/sdks/go/pkg/beam/validate.go
+++ b/sdks/go/pkg/beam/validate.go
@@ -71,8 +71,8 @@ func makeTypedefs(list []TypeDefinition) (map[string]reflect.Type, error) {
 		if !typex.IsUniversal(v.Var) {
 			return nil, errors.Errorf("type var %s must be a universal type", v.Var)
 		}
-		if !typex.IsConcrete(v.T) {
-			return nil, errors.Errorf("type value %s must be a concrete type", v.T)
+		if ok, err := typex.CheckConcrete(v.T); !ok {
+			return nil, errors.Wrapf(err, "type value %s must be a concrete type", v.T)
 		}
 		typedefs[v.Var.Name()] = v.T
 	}

